src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.Span.cs

```diff
@@ -57,9 +57,9 @@ private static object ParseCore(ReadOnlySpan<byte> utf8Json, Type returnType, Js
             var reader = new Utf8JsonReader(utf8Json, isFinalBlock: true, readerState);
             object result = ReadCore(returnType, options, ref reader);
 
-            readerState = reader.CurrentState;
-            if (readerState.BytesConsumed != utf8Json.Length)
+            if (reader.BytesConsumed != utf8Json.Length)
             {
+                readerState = reader.CurrentState;
```

Comment:
Actually, do we need to capture the reader state at all?
It looks to me all that's needed is `BytesConsumed`, `lineNumber,` and `bytePositionInLine`, all of which the reader has already.

Reply:
So would you want to make those `internal` and use them instead for exceptions?

Reply:
That, or I would actually pass the whole reader by ref to the `ThrowHelper`, like we do elsewhere and minimize method size for the non-exception path (and defer fetching the state):
https://github.com/dotnet/corefx/blob/b35cd0db472434cc5991f305de4557d51991b4a4/src/System.Text.Json/src/System/Text/Json/ThrowHelper.cs#L218-L228

Reply:
In any case, this can be done separately, and it isn't that critical. @steveharter, let me know if you have any dissenting thoughts on this.